src/Functions/if.cpp

```diff
@@ -1112,17 +1201,12 @@ class FunctionIf : public FunctionIfBase
 
         if (cond_const_col)
         {
-            if (arg_then.type->equals(*arg_else.type))
-            {
-                return cond_const_col->getValue<UInt8>()
-                    ? arg_then.column
-                    : arg_else.column;
-            }
+            UInt8 value = cond_const_col->getValue<UInt8>();
```

Comment:
This code is now much clearer, thanks!. Out of curiosity, what led you to change it in the PR? Did you find it when testing constant floats/decimals? Does is affect performance anyway (I guess it doesn't, but it doesn't hurt to ask).

Reply:
I found it could be simplified while reviewing the codes in if.cpp.  We use if frequently in Gluten for substrait scalar function parsing(e.g. https://github.com/oap-project/gluten/blob/main/cpp-ch/local-engine/Parser/scalar_function_parser/arrayElement.h). Every trivial similar optimization makes sense for Gluten, which I hope CH will also benefit from.  

However, I can't prove performance is improved after current changes..